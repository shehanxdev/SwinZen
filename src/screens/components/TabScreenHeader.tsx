import { View } from 'react-native';

import { Text } from '@sz/components';
import { tw } from '@sz/config';
import { TextVariant } from '@sz/constants';

export interface TabScreenHeaderProps {
  title: string;
}

export function TabScreenHeader({ title }: TabScreenHeaderProps) {
  return (
    <View style={tw`pt-8`}>
      <Text variant={TextVariant.SubTitle2SemiBold}>{title}</Text>
    </View>
  );
}
