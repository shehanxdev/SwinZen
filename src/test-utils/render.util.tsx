import { NavigationContext } from '@react-navigation/native';
import { render } from '@testing-library/react-native';
import React, { ReactNode } from 'react';
import Config from 'react-native-config';
import { Provider as PaperProvider } from 'react-native-paper';
import { Provider } from 'react-redux';

import { paperTheme } from '@sz/config';
import { ConfigService } from '@sz/services';
import { store } from '@sz/stores';

type RenderParams = Parameters<typeof render>;

export function renderWithProviders(ui: RenderParams[0], options?: RenderParams[1]) {
  ConfigService.setCallback(key => Config[key]);

  const BaseProviders = ({ children }: { children: ReactNode }) => (
    <Provider store={store}>
      <NavigationContext.Provider
        //@ts-ignore
        value={{
          isFocused: () => true,
          addListener: jest.fn(() => jest.fn()),
        }}>
        <PaperProvider theme={paperTheme}>{children}</PaperProvider>
      </NavigationContext.Provider>
    </Provider>
  );

  return render(ui, { wrapper: BaseProviders, ...options });
}
