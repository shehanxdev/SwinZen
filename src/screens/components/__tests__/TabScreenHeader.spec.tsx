import { renderWithProviders } from '@sz/test-utils';

import { TabScreenHeader } from '../TabScreenHeader';

describe('TabScreenHeader', () => {
  const getRenderedScreen = () => renderWithProviders(<TabScreenHeader title="Title" />);
  it('Should render', () => {
    const renderer = getRenderedScreen();
    const renderTree = renderer.toJSON();
    expect(renderTree).toMatchSnapshot();
  });
});
