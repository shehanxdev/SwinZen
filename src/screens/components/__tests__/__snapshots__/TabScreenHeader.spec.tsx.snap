// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabScreenHeader Should render 1`] = `
<View
  collapsable={false}
  pointerEvents="box-none"
  style={
    {
      "flex": 1,
    }
  }
>
  <View
    style={
      {
        "paddingTop": 32,
      }
    }
  >
    <Text
      style={
        [
          {
            "color": "#F8F9FA",
            "lineHeight": 28,
          },
          {
            "fontFamily": "SourceSansPro-SemiBold",
            "fontSize": 24,
            "fontWeight": "600",
            "textAlign": "center",
          },
        ]
      }
    >
      Title
    </Text>
  </View>
</View>
`;
